analyzers/src/SonarAnalyzer.CSharp/Rules/ClassWithOnlyStaticMember.cs

```diff
@@ -79,18 +80,26 @@ private static void CheckConstructors(SonarSymbolReportingContext context, IName
                 return;
             }
 
-            var reportMessage = string.Format(MessageFormatConstructor, utilityClass.IsSealed ? SyntaxConstants.Private : SyntaxConstants.Protected);
-
             foreach (var constructor in utilityClass.GetMembers()
                                                     .Where(IsConstructor)
                                                     .Where(symbol => ProblematicConstructorAccessibility.Contains(symbol.DeclaredAccessibility)))
             {
                 var syntaxReferences = constructor.DeclaringSyntaxReferences;
                 foreach (var syntaxReference in syntaxReferences)
                 {
-                    if (syntaxReference.GetSyntax() is ConstructorDeclarationSyntax constructorDeclaration)
+                    switch (syntaxReference.GetSyntax())
                     {
-                        context.ReportIssue(Diagnostic.Create(Rule, constructorDeclaration.Identifier.GetLocation(), reportMessage));
+                        case ConstructorDeclarationSyntax constructorDeclaration:
+                            var reportMessage = string.Format(MessageFormatConstructor, utilityClass.IsSealed ? SyntaxConstants.Private : SyntaxConstants.Protected);
+                            context.ReportIssue(Diagnostic.Create(Rule, constructorDeclaration.Identifier.GetLocation(), reportMessage));
+                            break;
+                        case ClassDeclarationSyntax classDeclaration when ClassDeclarationSyntaxWrapper.IsInstance(classDeclaration)
+                                && (ClassDeclarationSyntaxWrapper)classDeclaration is var classDeclarationWrapper
+                                && classDeclarationWrapper.ParameterList is { Parameters.Count: 0 }:
```

Comment:
Waiting for the `ParameterList()` extension method from #8209 to simplify this expression
https://github.com/SonarSource/sonar-dotnet/pull/8209/files#diff-0aed6645a137b426dc5eb96dcf45438091d2b6a5dba073c7c688407106ab9f35R54